js/ui/appDisplay.js

```diff
@@ -1089,6 +1116,13 @@ const AppSearchProvider = new Lang.Class({
     createResultObject: function (resultMeta) {
         let app = this._appSys.lookup_app(resultMeta['id']);
         return new AppIcon(app);
+    },
+
+    activateResult: function(appId) {
```

Comment:
I think the idea here is that the existing `activate()` method on `AppIcon` will be called no? Not sure you need this method.